### PR TITLE
[FIRRTL/ExpandWhens] Support vector types

### PR DESCRIPTION
**Pull request by uenoku**
This commit changes expandWhens to accept vector types under the condition that connections are expanded, i.e. all the connections are between ground types.
I changed `declareSinks` to handle vector types and diff is very small thanks to FieldRef implementation.

### CODE REVIEW

File: test/Dialect/FIRRTL/expand-whens.mlir [hunk 1]
```diff
@@ -405,4 +405,61 @@ firrtl.module @analog(out %analog : !firrtl.analog<1>) {
...
+
+// CHECK-LABEL: @vector_of_bundle
+firrtl.module @vector_of_bundle(in %p : !firrtl.uint<1>, out %ret: !firrtl.vector<bundle<a:uint<1>>, 1>) {
+  %0 = firrtl.subindex %ret[0] : !firrtl.vector<bundle<a:uint<1>>, 1>
+  %1 = firrtl.subfield %0(0) : (!firrtl.bundle<a:uint<1>>) -> !firrtl.uint<1>
+  %c1_ui1 = firrtl.constant 1 : !firrtl.uint<1>
+  %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  firrtl.connect %1, %c0_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
+  // CHECK: firrtl.connect %1, %c1_ui1 : !firrtl.uint<1>, !firrtl.uint<1>
```
Comment (seldridge) [inline test/Dialect/FIRRTL/expand-whens.mlir:463]: You could add a `CHECK-NOT` to these to make sure that this is removing the first connection. I think that would have to come _before_ the `CHECK` to make sure that it wasn't using the earlier one.

**Review by seldridge — APPROVED** [review]
Nice! I'm happy that this was so clean...

**Comment by youngar** [comment]
Looks great!  One more thing is that register's [self connect](https://github.com/llvm/circt/blob/main/lib/Dialect/FIRRTL/Transforms/ExpandWhens.cpp#L188) will need to be recursive as well, but maybe we can do that in a follow up commit if we want.

**Comment by uenoku** [comment]
> Looks great! One more thing is that register's self connect will need to be recursive as well, but maybe we can do that in a follow up commit if we want.

Thanks. It makes sense. Yeah, I missed that points. 
Because we need to expand connections of register's self connect if the register is a vector, I would like to support this after implementing expand connect. I'd like to just reject vector registers.

**Review by youngar — APPROVED** [review]

**Pull request merged**